starter-code/main.js

```diff
@@ -53,7 +54,188 @@ mongoClient.connect(`mongodb://localhost:27017/crunchbase`, (error, db) => {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+          case "9":
+            // 9. - How many employees has Facebook ?
+            db.collection('companies').find({ name: "Facebook" }, { name: 1, number_of_employees: 1, _id: 0 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+          case "10":
+            //10. - List the name of all the products of Facebook
+            db.collection('companies').find({ name: "Facebook" }, { name: 1, "products.name": 1, _id: 0 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+          case "11":
+            // 11. - List the people that are working at Facebook right now(check relationships field)
+            db.collection('companies').find({ name: "Facebook" }, { name: 1, "relationships.person.permalink": 1, _id: 0 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+          case "12":
+            //12. - List all the companies where "david-ebersman" has worked.
+            db.collection('companies').find({ "relationships.person.permalink": 'david-ebersman' }, { name: 1, _id: 0 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+
+          case "13":
+            // 13. - List by name the competitors of Facebook
+            db.collection('companies').find({ name: "Facebook" }, { name: 1, "competitions.competitor.permalink": 1, _id: 0 }).toArray((error, result) => {
```

Comment:
```suggestion
            db.collection('companies').find({ name: "Facebook" }, {"competitions.competitor.permalink": 1, _id: 0 }).map(doc => doc.competitions.map(elm => elm.competitor.name)).toArray((error, result) => {
```

Mongo tiene su propio map que te permite mapear los documentos retornados.